src/test/java/edu/wisc/my/messages/controller/MessagesControllerTest.java

```diff
@@ -38,7 +38,16 @@ public void siteIsUp() throws Exception {
    * building block towards richer tests of the application-as-running.
    */
   @Test
-  public void dataIsValid() {
-    this.messageReader.allMessages();
+  public void filteredMessageWorks() throws Exception {
+    mvc.perform(MockMvcRequestBuilders.get("/messages").accept(MediaType.APPLICATION_JSON))
```

Comment:
Should we mock the service here using when().thenReturn()?

Reply:
I don't understand this comment. `MessagesControllerTest` is an integration test driven by some magic annotations:

```
 @RunWith(SpringRunner.class)
 @SpringBootTest
 @AutoConfigureMockMvc
public class MessagesControllerTest {
```

As I understand it, in this kind of test, one doesn't use Mockito, that rather one would do that kind of mocking in a (unit) test where one is exercising more control over what (mock) objects are injected into the object under test?

That is, there's no service handy here available to be mocked?

But I'll readily admit I'm not very comfortable with these magic annotations and this "Boom! Have a whole MVC!" style of testing. That's much of why I introduced the more-traditional-unit-test-style Mockito driven unit test file alongside this test file.

Reply:
I wasn't sure what this was trying to test (integration vs unit).